pkg/types/ccipocr3/v2/generic_types.go

```diff
@@ -0,0 +1,190 @@
+package ccipocr3
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+)
+
+type TokenPrice struct {
+	TokenID UnknownEncodedAddress `json:"tokenID"`
+	Price   BigInt                `json:"price"`
+}
+
+func NewTokenPrice(tokenID UnknownEncodedAddress, price *big.Int) TokenPrice {
+	return TokenPrice{
+		TokenID: tokenID,
+		Price:   BigInt{price},
+	}
+}
+
+type GasPriceChain struct {
+	GasPrice BigInt        `json:"gasPrice"`
+	ChainSel ChainSelector `json:"chainSel"`
+}
+
+func NewGasPriceChain(gasPrice *big.Int, chainSel ChainSelector) GasPriceChain {
+	return GasPriceChain{
+		GasPrice: NewBigInt(gasPrice),
+		ChainSel: chainSel,
+	}
+}
+
+type SeqNum uint64
+
+func (s SeqNum) String() string {
+	return strconv.FormatUint(uint64(s), 10)
+}
+
+func NewSeqNumRange(start, end SeqNum) SeqNumRange {
+	return SeqNumRange{start, end}
+}
+
+// SeqNumRange defines an inclusive range of sequence numbers.
+type SeqNumRange [2]SeqNum
+
+func (s SeqNumRange) Start() SeqNum {
+	return s[0]
+}
+
+func (s SeqNumRange) End() SeqNum {
+	return s[1]
+}
+
+func (s *SeqNumRange) SetStart(v SeqNum) {
+	s[0] = v
+}
+
+func (s *SeqNumRange) SetEnd(v SeqNum) {
+	s[1] = v
+}
+
+// Limit returns a range limited up to n elements by truncating the end if necessary.
+// Example: [1 -> 10].Limit(5) => [1 -> 5]
+func (s *SeqNumRange) Limit(n uint64) SeqNumRange {
+	limitedRange := NewSeqNumRange(s.Start(), s.End())
+
+	numElems := s.End() - s.Start() + 1
+	if numElems <= 0 {
+		return limitedRange
+	}
+
+	if uint64(numElems) > n {
+		newEnd := limitedRange.Start() + SeqNum(n) - 1
+		if newEnd > limitedRange.End() { // overflow - do nothing
+			return limitedRange
+		}
+		limitedRange.SetEnd(newEnd)
+	}
+
+	return limitedRange
+}
+
+// Overlaps returns true if the two ranges overlap.
+func (s SeqNumRange) Overlaps(other SeqNumRange) bool {
+	return s.Start() <= other.End() && other.Start() <= s.End()
+}
+
+// Contains returns true if the range contains the given sequence number.
+func (s SeqNumRange) Contains(seq SeqNum) bool {
+	return s.Start() <= seq && seq <= s.End()
+}
+
+func (s SeqNumRange) String() string {
+	return fmt.Sprintf("[%d -> %d]", s[0], s[1])
```

Comment:
Seems like the SeqNumRange is inclusive of the endpoints, let's include that in a comment somewhere.

Any reason we use this specific arrow notation instead of a comma?